src/backend/distributed/transaction/relation_access_tracking.c

```diff
@@ -523,10 +525,12 @@ RecordParallelRelationAccess(Oid relationId, ShardPlacementAccessType placementA
 
 	/*
 	 * If a relation is partitioned, record accesses to all of its partitions as well.
-	 * We prefer to use PartitionedTableNoLock() because at this point the necessary
-	 * locks on the relation has already been acquired.
+	 *
+	 * TODO: We should use PartitionedTableNoLock() and PartitionTableNoLock() here
```

Comment:
I think in parallel access case, we don't need to be careful about performance, so I'm even up for removing the comment here.

The main idea for `NoLockPartitionedTable` is that we could call it too many times (as it used to be) and this could be a performance issue. For parallel queries, I think that's not going to be a performance issue for sure because we're calling `RecordParallelRelationAccess ` mostly once per distributed query/table.

Reply:
Removed the comment.